### PR TITLE
Comments: Add a query component for the `comment-counts` endpoint.

### LINKED ISSUE #20621: Comments: add comments count to data-layer
Let's wire up the new [comments count endpoint](https://github.com/Automattic/jetpack/pull/8202)! This will allow us create accurate pagers (the comment-list endpoint doesn't always return counts), and eventually show comment counts in our labels.

### PR DESCRIPTION
**Pull request by kwight**
Fixes #20621 .

This PR adds handling for the `comment-counts` endpoint to the data layer, along with a query component for calling it.

### CODE REVIEW

**Comment by kwight** [comment]
- [ ] Add `post_id` handling.
 - [ ] Add `COMMENT_COUNTS_ADD`.

File: client/components/data/query-site-comment-counts/index.jsx [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+		postId: PropTypes.number,
+		siteId: PropTypes.number,
+	};
+
+	componentDidMount() {
+		this.request();
+	}
+
+	componentDidUpdate() {
+		this.request();
```
Comment (gwwar) [inline client/components/data/query-site-comment-counts/index.jsx]: We should check if postId/site updated

File: client/state/data-layer/wpcom/sites/comment-counts/index.js [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+					postId,
+				},
+			},
+			action
+		)
+	);
+};
+
+const treeHandlers = {
+	[ COMMENT_COUNTS_REQUEST ]: [ dispatchRequest( fetchCommentCounts ) ],
```
Comment (gwwar) [inline client/state/data-layer/wpcom/sites/comment-counts/index.js]: We can use the new `dispatchRequestEx` which changes the signature of the handlers to return an action or action thunk.  (It'll eventually be renamed back to dispatchRequest once all instances are updated). 

See p4TIVU-8LU-p2 for a full example.

File: client/state/data-layer/wpcom/sites/comment-counts/index.js [hunk 3]
```diff
@@ -0,0 +1,71 @@
+/** @format */
+
+/**
+ * External dependencies
+ */
+
+// import { translate } from 'i18n-calypso';
+// import { flatMap, flatten, isArray, map } from 'lodash';
```
Comment (vindl) [inline client/state/data-layer/wpcom/sites/comment-counts/index.js]: Forgot to clean this up? :)

Comment (gwwar) [reply client/state/data-layer/wpcom/sites/comment-counts/index.js]: Thanks, I can spin that up in another PR

**Pull request merged**